### PR TITLE
Optional boundary assignment to cut surface

### LINKED ISSUE #9533: Add sideset tagging to ElementDeleterBase
### Description of the enhancement or error report
It would be useful to be able to assign a boundary ID to the cut surface exposed by element deleting classes (derived from `ElementDeleterBase`)

### Rationale for the enhancement or information for reproducing the error
Will make these classes more useful. BCs can be added to the holes.

### Identified impact
Added capability

### PR DESCRIPTION
**Pull request by dschwen**
Closes #9533

### CODE REVIEW

**Comment by fdkong** [comment]
I am interested in this PR.

File: test/tests/mesh_modifiers/block_deleter/BlockDeleterTest12.i [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    type = SubdomainBoundingBox
+    block_id = 1
+    bottom_left = '0.9 0.9 0'
+    top_right = '3.1 3.1 0'
+  [../]
+  [./delete]
+    type = BlockDeleter
+    block_id = 1
+    depends_on = mark
+    new_boundary = cut_surface
```
Comment (fdkong) [inline test/tests/mesh_modifiers/block_deleter/BlockDeleterTest12.i]: Could we assign an ID instead of boundary name?

Comment (dschwen) [inline test/tests/mesh_modifiers/block_deleter/BlockDeleterTest12.i]: Sure, that already works automagically.

**Comment by dschwen** [comment]
Here is an example of a custom class derived from `ElementDeleterBase` which deletes in an ellipsoidal region and snaps nodes to the geometric cut surface. Here assigning the boundary in the base class makes it real easy (as compared to additional mesh modifiers).

cut

**Comment by jwpeterson** [comment]
This is kind of cool but there are some very poor quality triangles around this ellipse...

**Comment by YaqiWang** [comment]
How do you 'snaps nodes to the geometric cut surface'? You have a particular modifier derived from the base?

**Comment by dschwen** [comment]
> there are some very poor quality triangles around this ellipse

Yeah, this is tangential to this PR, but you're right. I could displace nodes right outside the cut surface to improve the element quality (btw. those are QUAD9s).
It does not seem to have an adverse effect on the solve of the Stokes problem though.

**Comment by dschwen** [comment]
@YaqiWang, yes, this one: https://github.com/dschwen/CHiMaD_Hackathon/blob/master/src/meshmodifiers/CutHole.C

**Comment by YaqiWang** [comment]
Oh, a nice simple code.  BTW, if two erased nodes happen aligned with the centroid of the hole, your snapping will be screwed, right?

**Comment by dschwen** [comment]
Yeah, I guess. If no element is fully contained by the ellipsoid (only those get deleted!) _and_ a node is on the ellipse centroid, _then_ I probably get a division by zero. I might as well error out or return early if no elements end up getting deleted. Good catch!

**Comment by dschwen** [comment]
With new MeshModifier that moves nodes around to make element less bad:

cut2

**Comment by YaqiWang** [comment]
Nice picture!

**Review by fdkong — APPROVED** [review]

**Pull request merged**